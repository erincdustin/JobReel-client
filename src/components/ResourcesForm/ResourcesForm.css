.resources-form {
  grid-column-start: 4;
  grid-column-end: 7;
  border: 1px black solid;
  min-width: 300px;
  margin: 0 auto;
  padding: 5%;
  line-height: 150%;
  border-radius: 10px;
  text-align: center;
  background: #3a3846;
    color: white;
  box-shadow:  0.5em 0.5em 0.2em rgb(48, 39, 39);
  height: 100%;
}

#resources-description-input {
  width: 15em;
  border-radius: 10px;
}

#resource-type-input {
  width: 15em;
  border-radius: 10px;
}

@media screen and (max-width: 960px) {
  .resources-form {
    grid-column-start: 1;
    grid-column-end: 9;
  }
}