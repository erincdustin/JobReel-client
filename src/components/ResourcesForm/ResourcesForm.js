import React from 'react';
import JobReelContext from '../../context/JobReelContext';
import { Label, Input } from '../Form/Form';
import Button from '../Button/Button';
import './ResourcesForm.css';
import JobReelApiService from '../../services/jobreel-api-service';


export default class ResourcesForm extends React.Component {
  static contextType = JobReelContext;

  state = {
    error: null
  };

  handleSubmit = (e) => {
    e.preventDefault();
    this.setState({ error: null });
    const type = e.target.type.value;
    const title = e.target.title.value;
    const description = e.target.description.value;
    const date_added = new Date();
    const userInput = { user_id: this.context.user.id, type, title, description, date_added };
    JobReelApiService.submitResource(userInput)
      .then(res => {
        this.context.setResources([...this.context.resources, res]);
        this.context.setManualResourceAdd(false);
      })
      .catch(res => {
        this.setState({ error: res.error });
    })
        e.target.type.value = '';
        e.target.title.value = '';
        e.target.description.value = '';
  }

  renderTypeOptions = () => {
    const types = [
      'website',
      'book',
      'github repository',
      'magazine',
      'online publication',
      'podcast'
    ]
    return types.map((type,i) => {
      return <option key={i} value={type}>{type}</option>
    })
  }


  

  render() {
    const { error } = this.state;
    return (
      <div className='resources-form'>
        <div role='alert'>
          {error && <p>{error}</p>}
        </div>
        <form onSubmit={this.handleSubmit}>
          <div>
            <Label htmlFor='resource-type-input'>
              Type
            </Label>
            <br />
            <select name='type' id='resource-type-input'>
              {this.renderTypeOptions()}
            </select>
          </div>
          <div>
            <Label htmlFor='resource-title-input'>
              Title
            </Label>
            <br />
            <Input
              id='resource-title-input'
              name='title'
            />
          </div>
          <div>
            <Label htmlFor='resources-description-input'>
              Description
            </Label>
            <br/>
            <textarea
              id='resources-description-input'
              name='description'
            />
          </div>
          <Button onClick={() => this.context.setManualResourceAdd(false)} type='button'>Go Back</Button>
          <Button type='submit'>Submit</Button>
        </form>
      </div>
    )
  }
}