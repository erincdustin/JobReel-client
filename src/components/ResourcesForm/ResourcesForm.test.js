import React from 'react';
import ReactDOM from 'react-dom'
import ResourcesForm from './ResourcesForm'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('ResourcesForm smoke test', () => {
  it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<ResourcesForm />, div);
      ReactDOM.unmountComponentAtNode(div);
  })
})

describe('ResourcesForm snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(<ResourcesForm />)
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})