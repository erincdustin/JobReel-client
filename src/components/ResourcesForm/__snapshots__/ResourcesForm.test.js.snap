// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourcesForm snapshot test renders the UI as expected 1`] = `
<div
  className="resources-form"
>
  <div
    role="alert"
  />
  <form
    onSubmit={[Function]}
  >
    <div>
      <Label
        htmlFor="resource-type-input"
      >
        Type
      </Label>
      <br />
      <select
        id="resource-type-input"
        name="type"
      >
        <option
          key="0"
          value="website"
        >
          website
        </option>
        <option
          key="1"
          value="book"
        >
          book
        </option>
        <option
          key="2"
          value="github repository"
        >
          github repository
        </option>
        <option
          key="3"
          value="magazine"
        >
          magazine
        </option>
        <option
          key="4"
          value="online publication"
        >
          online publication
        </option>
        <option
          key="5"
          value="podcast"
        >
          podcast
        </option>
      </select>
    </div>
    <div>
      <Label
        htmlFor="resource-title-input"
      >
        Title
      </Label>
      <br />
      <ForwardRef
        id="resource-title-input"
        name="title"
      />
    </div>
    <div>
      <Label
        htmlFor="resources-description-input"
      >
        Description
      </Label>
      <br />
      <textarea
        id="resources-description-input"
        name="description"
      />
    </div>
    <ForwardRef
      onClick={[Function]}
      type="button"
    >
      Go Back
    </ForwardRef>
    <ForwardRef
      type="submit"
    >
      Submit
    </ForwardRef>
  </form>
</div>
`;
